tempto-core/src/main/java/com/teradata/tempto/internal/ReflectionInjectorHelper.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.teradata.tempto.internal;
+
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static java.util.Arrays.asList;
+
+public class ReflectionInjectorHelper
+{
+    public Object[] getMethodArguments(Injector injector, Method method)
+    {
+        if (isAnnotatedWithInject(method)) {
+            Parameter[] parameters = method.getParameters();
+            Annotation[][] parameterAnnotations = method.getParameterAnnotations();
+            Object[] methodArgs = new Object[parameters.length];
+            for (int i = 0; i < parameters.length; i++) {
+                methodArgs[i] = getInstanceForParameter(injector, parameters[i], asList(parameterAnnotations[i]));
+            }
+            return methodArgs;
+        }
+        else {
+            return new Object[] {};
+        }
+    }
+
+    private boolean isAnnotatedWithInject(Method method)
+    {
+        return method.getAnnotation(Inject.class) != null ||
+                method.getAnnotation(javax.inject.Inject.class) != null;
+    }
+
+    private Object getInstanceForParameter(Injector injector, Parameter parameter, List<Annotation> parameterAnnotation)
+    {
+        TypeLiteral type = TypeLiteral.get(parameter.getParameterizedType());
+        Optional<Annotation> namedAnnotation = parameterAnnotation.stream()
+                .filter(namedAnnotation())
+                .findFirst();
+
+        if (namedAnnotation.isPresent()) {
```

Comment:
`@Named` is just one of possibilities. Any annotation can be part of injection key.